### PR TITLE
nixos/dovecot: Switch systemd service type to notify

### PR DESCRIPTION
**Pull request by dminuoso**
@hrdinka @ajs124 

###### Motivation for this change

The dovecot systemd service uses type `simple`, which fails to report errors when dovecot fails to start during `systemctl start/restart` and reports success instead. This hides configuration bugs.

###### Things done

Switched systemd service to type `notify` which dovecot supports.

Also switched the `ExecStop` to `doveadm stop`. Before, we relied on whatever systemd does by default. Right now both end up sending SIGTERM to the master process, but I prefer we use the public interface. This also allows for future dovecot updates to do additional things for cleanup.

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/mail/dovecot.nix [hunk 1]
```diff
@@ -429,7 +429,9 @@ in
 
       startLimitIntervalSec = 60;  # 1 min
       serviceConfig = {
+        Type = "notify";
         ExecStart = "${dovecotPkg}/sbin/dovecot -F";
+        ExecStop = "${dovecotPkg}/sbin/doveadm stop";
```
Comment (ajs124) [inline nixos/modules/services/mail/dovecot.nix:434]: I'm not sure if this actually does what we want. Looking at the [relevant systemd documentation](https://www.freedesktop.org/software/systemd/man/systemd.service.html#ExecStop=), one thing sticks out: "it is usually not sufficient to specify a command for this setting that only asks the service to terminate (for example, by sending some form of termination signal to it), but does not wait for it to do so". Does doveadm do that?

**Comment by dminuoso** [comment]
Yes. See https://github.com/dovecot/core/blob/master/src/doveadm/doveadm-master.c#L70

**Comment by ajs124** [comment]
Hm. I'm not sure the ExecStop is better than just having systemd, tbh. Right now, the service gets a SIGTERM and if it fails to respond for long enough a SIGKILL. These signals, times between them etc can be configured the same way you can configure them for every other service that doesn't have `ExecStop=`. If `doveadm stop` were more than "send a SIGTERM and wait für 3 seconds" I'd be on board, but it isn't.

`Type = "notify";` I'm fine with though.

**Comment by dminuoso** [comment]
Agreed and PR updated.

This is ready for final review/merge then.

For future reference, if somebody wants to backport this, this PR depends on `c466620ad2c70caedd2bb12a56546ff05bf2f63c`

**Pull request merged**